File: net/YapayGatewayLib/model/ItemData.cs
using System;
using Newtonsoft.Json;

namespace YapayGatewayLib.Model
{
    public class ItemData
    {
	[JsonProperty(PropertyName = "codigoProduto")]
	string productCode{get;set;}

	[JsonProperty(PropertyName = "codigoCategoria")]
	string productCategory{get;set;}

	[JsonProperty(PropertyName = "nomeProduto")]
	string productName{get;set;}

	[JsonProperty(PropertyName = "quantidadeProduto")]
    int productAmount{get;set;}

	[JsonProperty(PropertyName = "valorUnitarioProduto")]
	long productUnitaryValue{get;set;}

	[JsonProperty(PropertyName = "nomeCategoria")]
	string categoryName{get;set;}

    }
}

File: java/yapay-gateway-client/src/main/java/br/com/yapay/gateway/model/ItemData.java
package br.com.yapay.gateway.model;

import java.math.BigDecimal;

import com.google.gson.annotations.SerializedName;

/**
 * Class describing cart items
 * 
 * @author Adriano Santos
 *
 */
public class ItemData {

	@SerializedName("codigoProduto")
	private String productCode;

	@SerializedName("codigoCategoria")
	private String productCategory;

	@SerializedName("nomeProduto")
	private String productName;

	@SerializedName("quantidadeProduto")
	private Integer productAmount;

	@SerializedName("valorUnitarioProduto")
	private Long productUnitaryValueLong;

	@SerializedName("nomeCategoria")
	private String categoryName;

	public String getProductCode() {
		return productCode;
	}

	public void setProductCode(String productCode) {
		this.productCode = productCode;
	}

	public String getProductCategory() {
		return productCategory;
	}

	public void setProductCategory(String productCategory) {
		this.productCategory = productCategory;
	}

	public String getProductName() {
		return productName;
	}

	public void setProductName(String productName) {
		this.productName = productName;
	}

	public Integer getProductAmount() {
		return productAmount;
	}

	public void setProductAmount(Integer productAmount) {
		this.productAmount = productAmount;
	}

	Long getProductUnitaryValueLong() {
		return productUnitaryValueLong;
	}

	void setProductUnitaryValueLong(Long productUnitaryValue) {
		this.productUnitaryValueLong = productUnitaryValue;
	}

	public void setProductUnitaryValue(BigDecimal productUnitaryValue) {
		this.productUnitaryValueLong = productUnitaryValue == null ? null
				: productUnitaryValue.multiply(new BigDecimal(100)).longValue();
	}

	public String getCategoryName() {
		return categoryName;
	}

	public void setCategoryName(String categoryName) {
		this.categoryName = categoryName;
	}

}
